cmd/main.go

```diff
@@ -1,9 +1,49 @@
 package main
 
 import (
-	"github.com/dsauerbrun/cragcast/api"
+	"log"
+
+	"github.com/dsauerbrun/cragcast/pkg/http"
+	"github.com/spf13/cobra"
 )
 
+var bindHost string
+var port int
+
+func init() {
+	serveCmd.Flags().StringVar(&bindHost, "bind-host", "127.0.0.1", "The address to bind to. To accept all connections, use 0.0.0.0.")
+	serveCmd.Flags().IntVar(&port, "port", api.DefaultPort, "The port to listen on.")
+
+	rootCmd.CompletionOptions.DisableDefaultCmd = true
+	rootCmd.AddCommand(serveCmd)
```

Comment:
i see that init() is adding the `serve` command to the cli which the makefile utilizes... however, I can't see anywhere that init() is actually being called. What's going on here with this new main.go?

Reply:
nvm, i just read that this is a go thing and that the init function is always called first when a package is initialized

Reply:
Exactly. Generally `init` is a practice to stay way from, but it has some justified use-cases, such as cmd-line initialization (e.g. flags).